### ISSUE TITLE
SignatureMarkTextHL cannot really be a function reference

### ISSUE THREAD

**Issue by rkennedy** [kshenoy/vim-signature#91]
The [documentation for `SignatureMarkTextHL`][1] says it can be set to a function reference, so I expected to be able to set the variable like so:

```vim
function RKGetHLGroup(lineno)
...
endfunction
let g:SignatureMarkTextHL = function('RKGetHLGroup')
```

That causes errors, though:

```text
Error detected while processing function signature#sign#Refresh..signature#sign#Place..signature#sign#RefreshLine:
line   20:
E729: using Funcref as a String
E729: using Funcref as a String
E729: using Funcref as a String
E729: using Funcref as a String
```

The line 20 referenced refers to this code:

```vim
let l:SignatureMarkTextHL = eval( g:SignatureMarkTextHL )
```

It uses `eval`, but if it were really a function reference (as created with [`function()`][2]), it should use `call` instead.

Using `eval` is fine (although brittle as far as APIs go), but then the documentation shouldn't say "function reference," and it should probably mention some expressions that might be relevant in the evaluated string, such as `a:lnum`. If function references really are supposed to work, then both the code and the documentation need to change.

I think I'm comfortable contributing the changes myself, but I'd like some advice on which direction you'd prefer.

[1]: https://github.com/kshenoy/vim-signature/blob/master/doc/signature.txt#L192-L198
[2]: http://vimdoc.sourceforge.net/htmldoc/eval.html#function()

**Comment by kshenoy** [comment]
Hmm, thanks for pointing that out. Function reference is really a bad choice of words.

In the common case `g:SignatureMarkTextHL` will point to a highlight group. The eval was added later on as an enhancement. 

I could use a combination of strings for explicit highlight groups and funcrefs for dynamic highlighting and use [`type()`](http://vimdoc.sourceforge.net/htmldoc/eval.html#type()) to distinguish between them. However, I'm not sure how we would pass in arguments to `call()`

**Comment by rkennedy** [comment]
It turns out that `call()` isn't even needed. Function references can be called just like ordinary functions. Pull request #92 uses `type()` to detect strings and uses `eval()` for them, so all existing configurations should continue to work fine. (I browsed the first few dozen GitHub search results for `SignatureMarkTextHL` and didn't see anyone relying on anything other than `a:lnum` in their expressions, so the one argument should be sufficient.)

If you *wanted* to use `call`, it would go like this:

```vim
call(a:expression, [a:lnum])
```

That is, the second parameter to `call` is the list of function arguments.

**Comment by kshenoy** [comment]
Fair enough. Just the line number should probably suffice. 

Thanks for the detailed doc write-up

**Comment by kshenoy** [comment]
The code in `after/plugin/signature_gitgutter.vim` did not get updated in this pull and was corrected in a18b193.

Mentioning here for cross-reference
